ui/src/Dashboard/Value.jsx

```diff
@@ -0,0 +1,469 @@
+import assign from "lodash/assign";
+import cloneDeep from "lodash/cloneDeep";
+import without from "lodash/without";
+import moment from 'moment';
+
+import {
+  Button,
+  Card,
+  Col,
+  Row,
+  Space,
+  Switch,
+  Divider,
+  Popconfirm,
+  message,
+  Input,
+} from 'antd';
+import { useEffect, useState } from 'react';
+import {
+  CopyOutlined,
+} from '@ant-design/icons';
+
+import './Value.less';
+
+import {
+  ValueContext,
+  useValueState,
+  useProject,
+} from './context';
+import { ValueConditions } from './ValueConditions';
+import { TYPES, KIND_TO_TYPE, KIND, TYPE_TO_KIND } from './constants';
+import { useValueActions } from './actions';
+
+
+const ResetButton = ({ onClick, disabled }) => {
+  const style = disabled ? {} : { background: '#f90', color: '#fff' };
+  return (
+    <Button onClick={onClick} disabled={disabled} style={style}>
+      reset
+    </Button>
+  );
+}
+
+const CancelButton = ({ onClick, disabled }) => {
+  const style = disabled ? {} : { background: '#ff4d00', color: '#fff' };
+  return (
+    <Button onClick={onClick} style={style} disabled={disabled}>
+      cancel
+    </Button>
+  );
+}
+const DeleteButton = ({ onClick }) => {
+  const { name } = useValueState()
+
+  function confirm() {
+    onClick();
+    message.success(`Value ${name} deleted`);
+  }
+
+  return (
+    <Popconfirm
+      title="Are you sure to delete this value?"
+      onConfirm={confirm}
+      okText="Yes"
+      cancelText="No"
+    >
+      <Button type="primary" danger>
+        delete
+      </Button>
+    </Popconfirm>
+  );
+}
+
+const Buttons = ({ onReset, onCancel, onSave, onDelete, onToggle, onValueOverrideChange }) => {
+  const { dirty, overridden, enabled, value_override } = useValueState();
+  return (
+    <Row align='middle'>
+      <Col span={3}>
+        <Switch
+          onChange={onToggle}
+          checkedChildren="ON"
+          unCheckedChildren="OFF"
+          checked={enabled}
+        />
+      </Col>
+      <Col span={4}>
+        {overridden ?
+          <span style={{ color: 'green' }}>overriden</span> : 'default'}
+      </Col>
+      <Col span={6} offset={8}>
+        <Space size={2}>
+          <ResetButton onClick={onReset} disabled={!overridden}>
+            reset
+          </ResetButton>
+          <Button onClick={onSave} type="primary" disabled={!dirty}>
+            apply
+          </Button>
+          <CancelButton onClick={onCancel} disabled={!dirty}/>
+          <DeleteButton onClick={onDelete}/>
+        </Space>
+      </Col>
+      <Col span={8} offset={16} style={{ marginTop: 20 }}>
+        <Space size={0}>
+          <div>Current value</div>
+          <Input
+              value={value_override}
+              disabled={!enabled}
+              size={"middle"}
+              style={{ width: 165 }}
+              onChange={onValueOverrideChange}
+          />
+        </Space>
+      </Col>
+    </Row>
+  );
+}
+
+function copyToClipboard(text, msg) {
```

Comment:
Ideally this must go to some `utils.js` or `utils/clipboard.js`.

Same for `function _replace(array, value, newValue)` bellow since this code is a duplicate from Flag.jsx